### PR TITLE
Initialise package

### PR DESCRIPTION
**Pull request by rmbielby**
# Brief overview of changes

This PR sets the eesyapi repo up as a package with the base documentation (largely following dfeR and dfeshiny) and a first function: `api_url()`, which is used to render the url for a given API endpoint.

## Why are these changes being made?

API users working in R need a set of helper functions to smooth access to the EES API and this lays the groundwork for us to create those functions.

## Detailed description of changes

Setting up the package and adding a first function:
- Created a basic readme for the repo
- Copied across standard contributor guidelines, code of conduct etc from `dfeshiny`
- Initiated pkgdown to render documentation to github.io
- `api_url()` renders a url for an endpoint based on a set of parameters, including endpoint and dataset_id
- created a basic set of tests for `api_url()`
- set up workflows following what we have on dfeshiny

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+---
+
+## Describe the bug
+A clear and concise description of what the bug is.
+
+## How to reproduce
+Steps to reproduce the behaviour:
+1. Go to '...'
+2. Click on '....'
```
Comment (cjrace) [inline .github/ISSUE_TEMPLATE/bug_report.md:16]: I know you probably just copied this, though for this package we're unlikely to ever have people clicking and scrolling so could change to be more generic, e.g.

> 1. '...'
> 2. '...'
> 3. '...'

File: .github/workflows/lintr.yml
Comment (cjrace) [inline .github/workflows/lintr.yml:1]: This is different to the lint.yaml we have on dfeR - intentional? https://github.com/dfe-analytical-services/dfeR/blob/main/.github/workflows/lint.yaml

File: LICENSE
Comment (cjrace) [inline LICENSE:1]: Is this file still needed? Might be able to just delete

File: R/eesapi_url.R [hunk 2]
```diff
@@ -0,0 +1,78 @@
+#' eesapi_url
+#'
+#' @param api_version EES API version
+#' @param endpoint Name of endpoint, can be get-summary, get-meta, get-data, query-data
```
[the remark this answers is not in the document]
Comment (cjrace) [reply R/eesapi_url.R:4]: Safe to break the line in param descriptions so break this over two lines

File: R/eesapi_url.R [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+#' @param dataset_id ID of dataset to be connected to
+#' @param dataset_version Verson of dataset to be connected to
+#'
+#' @return The EES URL for connecting to the API
```
Comment (cjrace) [inline R/eesapi_url.R:8]: Should we be specific that it's a string that's returned, e.g.  `A string containing the URL for connecting to the EES API`?

File: R/eesapi_url.R [hunk 4]
```diff
@@ -0,0 +1,91 @@
+#' eesapi_url
+#'
+#' @param api_version EES API version
+#' @param endpoint Name of endpoint, can be get-summary, get-meta, get-data, query-data
+#' @param dataset_id ID of dataset to be connected to
+#' @param dataset_version Verson of dataset to be connected to
```
Comment (cjrace) [inline R/eesapi_url.R:6]: Typo - Version / Verson

Comment (cjrace) [inline R/eesapi_url.R:6]: data set is technically two words when in a sentence I think? (or even if not, it will be better to agree a convention now)

File: R/eesapi_url.R [hunk 5]
```diff
@@ -0,0 +1,91 @@
+#' eesapi_url
+#'
+#' @param api_version EES API version
+#' @param endpoint Name of endpoint, can be get-summary, get-meta, get-data, query-data
```
Comment (cjrace) [inline R/eesapi_url.R:4]: I'd put these examples in quotes as they would be in the function

File: R/eesapi_url.R [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        paste(
+          "You have entered an invalid endpoint, this should one of:",
+          "get-summary, get-meta, get-data or query-data"
+        )
+      )
+    }
+  }
+
+  is_valid_dataset_info <- function(dataset_id, dataset_version) {
+    !is.null(dataset_id) & (is.numeric(dataset_version) | is.null(dataset_version))
```
[the remark this answers is not in the document]
Comment (cjrace) [reply R/eesapi_url.R:51]: Had a few issues with the | operator and long lines / line breaks styling weird with styler, so if that's a pain when shortening this, I would be tempted to rewrite to avoid stringing conditions with | or &

File: R/eesapi_url.R [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+      ""
+    ),
+    ifelse(
+      endpoint %in% c("get-summary", "get-meta", "query-data"),
+      paste0("/", gsub("get-|-data", "", endpoint)),
+      ""
+    ),
+    ifelse(
+      !is.null(dataset_version),
+      paste0("?data-version=", dataset_version),
```
Comment (cjrace) [inline R/eesapi_url.R:87]: dev is down currently so I can't double check, but I think the data set version is camel rather than kebab, so `?dataSetVersion=`?

File: README.md [hunk 8]
```diff
@@ -1,2 +1,51 @@
...
+# <a href="https://dfe-analytical-services.github.io/eesyapi/">eesyapi</a>
+
+<!-- badges: start -->
+[![R-CMD-check](https://github.com/dfe-analytical-services/eesyapi/actions/workflows/R-CMD-check.yaml/badge.svg)](https://github.com/dfe-analytical-services/eesyapi/actions/workflows/R-CMD-check.yaml)
+[![Codecov test coverage](https://codecov.io/gh/dfe-analytical-services/eesyapi/branch/main/graph/badge.svg)](https://app.codecov.io/gh/dfe-analytical-services/eesyapi?branch=main)
+[![Lifecycle: experimental](https://img.shields.io/badge/lifecycle-experimental-orange.svg)](https://lifecycle.r-lib.org/articles/stages.html#experimental)
+<!-- badges: end -->
+
+An R package to support analysts in connecting to and processing data from 
+the DfE's Explore Education Statistics API endpoints.
```
Comment (cjrace) [inline README.md:10]: lower case explore education statistics

File: README.md [hunk 9]
```diff
@@ -1,2 +1,51 @@
...
+Otherwise:
+
+``` r
+# install.packages("devtools")
+devtools::install_github("dfe-analytical-services/eesyapi")
+```
+
+## Potential errors when installing
+
+If you get `ERROR [curl: (22) The requested URL returned error: 401]`, and don't
```
Comment (cjrace) [inline README.md:34]: Worth making it clear this is specifically aimed at analysts within DfE?

**Review by cjrace — CHANGES_REQUESTED** [review]
Left a few comments, all very minor / quick ones. Couple of lintr things flagged too.

Also noticed this note - assuming it's because httr is in there ready for when we do use it and it's just giving a note because we're not using it yet, so you can ignore it.

image

File: R/eesapi_url.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (rmbielby) [reply R/eesapi_url.R:4]: Fixed

File: R/eesapi_url.R (hunk 6 above)
[the remark this answers is not in the document]
Comment (rmbielby) [reply R/eesapi_url.R:52]: Added a new line

File: tests/testthat/test-eesapi_url.R [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+  )
+  expect_error(
+    eesapi_url(endpoint = 'query', dataset_id = "dummy:dataset:id")
+  )
+  expect_error(
+    eesapi_url(endpoint = 'query-data')
+  )
+  expect_equal(
+    eesapi_url(endpoint = 'query-data', dataset_id = "dummy:dataset:id"),
+    "https://dev.statistics.api.education.gov.uk/api/v1.0/data-sets/dummy:dataset:id/query"
```
[the remark this answers is not in the document]
Comment (rmbielby) [reply tests/testthat/test-eesapi_url.R:17]: Widened the character limit

File: .github/workflows/lintr.yml
Comment (rmbielby) [reply .github/workflows/lintr.yml:1]: Not intentional, but keeping as is following discussion on helping keeping us on the straight and narrow by flagging things directly in the PR comments.

File: LICENSE
Comment (rmbielby) [reply LICENSE:1]: I've removed it for now, assuming all we actually need is the LICENSE.md file

File: R/eesapi_url.R (hunk 3 above)
Comment (rmbielby) [reply R/eesapi_url.R:9]: Fair, changed as suggested.

File: R/eesapi_url.R (hunk 7 above)
Comment (rmbielby) [reply R/eesapi_url.R:88]: Yep, it was down for me as well, it's working now have been able to double check and updated to dataSetVersion

File: README.md (hunk 8 above)
Comment (rmbielby) [reply README.md:10]: Updated

File: README.md (hunk 9 above)
Comment (rmbielby) [reply README.md:34]: Done

File: LICENSE
Comment (rmbielby) [reply LICENSE:1]: That broke the devtools checks actually, so I've added it back in. Don't think this is a high priority to clean up right now, so leaving looking into how it should be done for now.

Comment (cjrace) [reply LICENSE:1]: Looks like you might need that for an MIT licence anyway actually, happy to leave

**Review by cjrace — APPROVED** [review]
